backend/cmm_helpers.ml

```diff
@@ -2262,6 +2271,20 @@ let bbswap bi arg dbg =
       [arg],
       dbg)
 
+let bbswap bi arg dbg =
+  let bitwidth =
+    match (bi : Primitive.boxed_integer) with
+    | Pnativeint -> size_int * 8
+    | Pint32 -> 32
+    | Pint64 -> 64
+  in
+  let op = Cbswap { bitwidth } in
+  if (bi = Primitive.Pint64 && size_int = 4) ||
+     not (Proc.operation_supported op) then
+    bbswap bi arg dbg
+  else
+    Cop (op,[arg],dbg)
+
 let bswap16 arg dbg =
   (Cop(Cextcall { func = "caml_bswap16_direct";
                   builtin = false;
```

Comment:
They were chosen conservatively, L2246 has a comment about it, I should put `if_operation_supported` before the comment.

Reply:
Well, the comment does not really say why,
and `ty_args` looks wrong even if we keep
effects and coeffects to their conservative
defaults.

Reply:
An empty list means "all arguments have default type XInt", see `emit_extcall_args` in `Selectgen`.

Reply:
Ah, yes, sorry; I forgot that convention.

Reply:
Back to the [co]effects, the primitives declarations
for "swap32" in the OCaml sources appears to be
precise. Are there scenarios in which the discrepancy
could be surprising to the user?

Reply:
The "slow" path, where we call "direct_bswap" is not as precise as it could be, in the sense that there are no effects or coeffects. The "fast" path, in which `Cmm_helpers` returns `Cbswap` operation, is precise (see `is_simple_expr` and `effects_of` in `Selection`).

Reply:
(Yes, that was my understanding; I cannot come up
with a scenario where that would be an issue, but I
am not sure such a scenario does not exist.)